File: frontend/src/lib/types/disbursement.ts
export interface DisbursementListItem {
  id: string;
  disbursement_reference: string;
  pfms_batch_reference: string;
  award_id: string;
  award_number: string;
  project_id: string;
  project_title: string;
  parcel_id: string;
  khasra_number: string;
  owner_id: string;
  owner_name: string;
  masked_bank_account: string;
  masked_ifsc: string;
  amount_inr: number | string;
  payment_method: string;
  payment_status: string;
  bank_utr_number?: string | null;
  disbursed_at?: string | null;
  created_at: string;
}

export interface DisbursementDetail {
  id: string;
  disbursement_reference: string;
  pfms_batch_reference: string;
  payment_workflow_label: string;
  award_id: string;
  award_number: string;
  award_amount_inr: number | string;
  project_id: string;
  project_title: string;
  project_code: string;
  district_name: string;
  state_name: string;
  parcel_id: string;
  khasra_number: string;
  owner_id: string;
  owner_name: string;
  relative_name?: string | null;
  masked_bank_account: string;
  masked_ifsc: string;
  bank_name: string;
  social_category: string;
  is_kyc_verified: boolean;
  amount_inr: number | string;
  payment_method: string;
  payment_status: string;
  bank_utr_number?: string | null;
  disbursed_at?: string | null;
  failure_reason?: string | null;
  remarks?: string | null;
  processed_by_user_name?: string | null;
  created_at: string;
  updated_at: string;
}

export interface FinancialReconciliationSummary {
  reference_id: string;
  reference_title: string;
  total_assessed_inr: number;
  total_awarded_inr: number;
  total_disbursed_inr: number;
  outstanding_inr: number;
  disbursement_percent: number;
  total_beneficiaries_count: number;
  disbursed_beneficiaries_count: number;
  pending_beneficiaries_count: number;
  reconciliation_status: string;
}

export interface DisbursementFilterParams {
  project_id?: string;
  award_id?: string;
  payment_status?: string;
  search?: string;
  page?: number;
  page_size?: number;
}

File: backend/app/schemas/disbursement.py
import uuid
from datetime import datetime
from decimal import Decimal
from typing import Optional, List
from pydantic import BaseModel, ConfigDict, Field


class DisbursementInitiateBatch(BaseModel):
    award_id: uuid.UUID
    parcel_ids: Optional[List[uuid.UUID]] = Field(default=None, description="Optional specific parcels; defaults to all parcels under the award")
    pfms_batch_reference: Optional[str] = None
    remarks: Optional[str] = "PFMS-compatible payment batch initiated for beneficiary direct benefit transfer."


class DisbursementProcessSimulationRequest(BaseModel):
    """Simulate PFMS bank response (Credit or Failure)."""
    target_status: str = Field(default="DISBURSED", description="DISBURSED or FAILED")
    bank_utr_number: Optional[str] = None
    failure_reason: Optional[str] = None


class DisbursementListItem(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: uuid.UUID
    disbursement_reference: str
    pfms_batch_reference: str
    award_id: uuid.UUID
    award_number: str
    project_id: uuid.UUID
    project_title: str
    parcel_id: uuid.UUID
    khasra_number: str
    owner_id: uuid.UUID
    owner_name: str
    masked_bank_account: str
    masked_ifsc: str
    amount_inr: Decimal
    payment_method: str = "PFMS_DBT"
    payment_status: str
    bank_utr_number: Optional[str] = None
    disbursed_at: Optional[datetime] = None
    created_at: datetime


class DisbursementDetail(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: uuid.UUID
    disbursement_reference: str
    pfms_batch_reference: str
    payment_workflow_label: str = "PFMS-Compatible / Simulated Payment Workflow"
    award_id: uuid.UUID
    award_number: str
    award_amount_inr: Decimal
    project_id: uuid.UUID
    project_title: str
    project_code: str
    district_name: str
    state_name: str
    parcel_id: uuid.UUID
    khasra_number: str
    owner_id: uuid.UUID
    owner_name: str
    relative_name: Optional[str] = None
    masked_bank_account: str
    masked_ifsc: str
    bank_name: str
    social_category: str
    is_kyc_verified: bool
    amount_inr: Decimal
    payment_method: str
    payment_status: str
    bank_utr_number: Optional[str] = None
    disbursed_at: Optional[datetime] = None
    failure_reason: Optional[str] = None
    remarks: Optional[str] = None
    processed_by_user_name: Optional[str] = None
    created_at: datetime
    updated_at: datetime


class FinancialReconciliationSummary(BaseModel):
    """Authoritative project or award level financial reconciliation."""
    reference_id: uuid.UUID
    reference_title: str
    total_assessed_inr: Decimal
    total_awarded_inr: Decimal
    total_disbursed_inr: Decimal
    outstanding_inr: Decimal
    disbursement_percent: Decimal
    total_beneficiaries_count: int
    disbursed_beneficiaries_count: int
    pending_beneficiaries_count: int
    reconciliation_status: str = "BALANCED"
